### PR TITLE
Return the tagValue of sorted tag

### PR DESCRIPTION
**Pull request by ButterBright**

### CODE REVIEW

File: api/proto/banyandb/database/v1/schema.proto [hunk 1]
```diff
@@ -156,8 +156,10 @@ message IndexRule {
   }
   // type is the IndexType of this IndexObject.
   Type type = 3 [(validate.rules).enum.defined_only = true];
+  // sorted indicates whether the index is for sorting.
+  bool sorted = 4;
```
Comment (wu-sheng) [inline api/proto/banyandb/database/v1/schema.proto:160]: Adding the flag(4) would make incompatible change.

The good practice should be the largest index for a new field.

Comment (wu-sheng) [reply api/proto/banyandb/database/v1/schema.proto:160]: You should move `sorted` to 7, and accept the value absent with a default value, which could keep the compatibility.

**Comment by wu-sheng** [comment]
Which scenario and how would the client uses this new return time?

**Comment by ButterBright** [comment]
When performing stream sorting, the tagValue will be used for merge sort, ensuring that not too many elements are read out.

File: api/proto/banyandb/database/v1/schema.proto (hunk 1 above)
Comment (hanahmily) [inline api/proto/banyandb/database/v1/schema.proto]: Please set the default value to 'true' for compatibility.

**Review by hanahmily — CHANGES_REQUESTED** [review]

**Comment by wu-sheng** [comment]
BTW @hanahmily @ButterBright As we are adding more features into server side, and we plan to keep the compatibility as much as possible with OAP, we should run a matrix e2e testing.

OAP should be set for a version matrix. Such as in this case, we should support the version is 10.0.1 and latest(not latest commit, just another commit ID with `latest`name). Then we will know the version compatible list for the latest BanyanDB server.

**Review by hanahmily — CHANGES_REQUESTED** [review]
You must also check whether the index rule supports sorting in the stream query analyzer.

**Review by hanahmily — APPROVED** [review]

**Pull request merged**